evalml/automl/automl_search.py

```diff
@@ -1148,12 +1158,14 @@ def _should_continue(self):
 
         num_pipelines = self._num_pipelines()
 
-        # check max_time and max_iterations
+        # check max_time, max_iterations and max_batches
         elapsed = time.time() - self._start
         if self.max_time and elapsed >= self.max_time:
             return False
         elif self.max_iterations and num_pipelines >= self.max_iterations:
             return False
+        elif self.max_batches and self._get_batch_number() > self.max_batches:
```

Comment:
This logic would only be invoked for `DefaultAlgorithm` as `max_iterations` is set when it is None when using `IterativeAlgorith`.